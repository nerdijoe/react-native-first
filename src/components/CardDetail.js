import React from 'react'
import {
  View,
  Text
} from 'react-native'

class CardDetail extends React.Component {

  render() {
    return (
      <View>
        <Text>Card Detail</Text>
      </View>
    )
  }
}

export default CardDetail
