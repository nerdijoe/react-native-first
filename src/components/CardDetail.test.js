import React from 'react'
import renderer from 'react-test-renderer'

import CardDetail from './CardDetail'

describe('CardDetail', () => {
  let card = {
    "_id":"58d176a56d804500104b1248",
    "idName":"arrows",
    "rarity":"Common",
    "type":"Spell",
    "name":"Arrows",
    "description":"Arrows pepper a large area, damaging everyone hit. Reduced damage to Crown Towers.",
    "arena":0,
    "elixirCost":3,
    "order":1,
    "__v":0
  }

  it('renders without crashing', ()=> {
    const tree = renderer.create(<CardDetail card={card}/>)
    expect(tree).toMatchSnapshot()
  })
})
