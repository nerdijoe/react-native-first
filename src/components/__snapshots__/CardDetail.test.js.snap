// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardDetail renders without crashing 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flex": 1,
          "flexDirection": "column",
          "flexWrap": "nowrap",
          "marginBottom": 30,
          "marginTop": 40,
          "padding": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 25,
            "fontWeight": "bold",
          }
        }
      >
        Arrows
      </Text>
      <Image
        source={
          Object {
            "uri": "http://www.clashapi.xyz/images/cards/arrows.png",
          }
        }
        style={
          Object {
            "height": 131.5,
            "margin": 10,
            "padding": 10,
            "width": 106,
          }
        }
      />
      <View
        style={
          Object {
            "borderRadius": 8,
            "borderStyle": "solid",
            "borderWidth": 1,
            "margin": 10,
            "padding": 10,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 10,
              "fontWeight": "bold",
              "marginTop": 5,
            }
          }
        >
          Description
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Arrows pepper a large area, damaging everyone hit. Reduced damage to Crown Towers.
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 10,
              "fontWeight": "bold",
              "marginTop": 5,
            }
          }
        >
          Arena
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          0
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 10,
              "fontWeight": "bold",
              "marginTop": 5,
            }
          }
        >
          Rarity
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Common
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 10,
              "fontWeight": "bold",
              "marginTop": 5,
            }
          }
        >
          Type
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Spell
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 10,
              "fontWeight": "bold",
              "marginTop": 5,
            }
          }
        >
          Elixir Cost
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          3
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
